kernel/src/thread/kernel_thread.rs

```diff
@@ -67,6 +41,32 @@ impl KernelThreadExt for Thread {
     }
 }
 
+/// Creates a new task of kernel thread, **NOT** run the thread.
+pub fn create_new_kernel_task(mut thread_options: ThreadOptions) -> Arc<Task> {
+    let task_fn = thread_options.take_func();
+    let thread_fn = move || {
+        task_fn();
+        let current_thread = current_thread!();
```

Comment:
One interesting observation is that if we create a new strong reference to the current thread at the beginning of the closure, then we won't have to worry about a kernel thread being dropped when its code is sleeping.

```rust
    let thread_fn = move || {
        // Ensure that at least one instance of `Arc<Thread>` exists
        // as long as the thread starts executing.
        let current_thread = current_thread!();
        task_fn();
        // Ensures the thread is exit
        current_thread.exit();
    };
```

Therefore, you don't need to change from `Thread` holding `Task` into `Task` holding `Thread` simply because you need to prevent `Thread` from being dropped. You can decide who holds the strong reference to who from other angles such as performance.

Reply:
Yes, from the performance perspective, I think the new version is also better. For example, the `current!` macro cost decreases from 38ns to 27ns(by avoiding an redundant Arc::clone)